datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -276,20 +266,6 @@ impl RecordBatchStream for CoalesceBatchesStream {
     }
 }
 
-/// Concatenates an array of `RecordBatch` into one batch
-pub fn concat_batches(
-    schema: &SchemaRef,
-    batches: &[RecordBatch],
-    row_count: usize,
-) -> ArrowResult<RecordBatch> {
-    trace!(
-        "Combined {} batches containing {} rows",
-        batches.len(),
-        row_count
-    );
-    arrow::compute::concat_batches(schema, batches)
```

Comment:
The existing function simply wraps `concat_batches` in arrow and logs some information which I don't think is critical. Note that the `row_count` is only used for the logging.  If reviewers think the `trace!` is valuable, I will put it back (at the callsites)